mwclient/client.py

```diff
@@ -612,6 +616,380 @@ def get_token(self, type, force=False, title=None):
+             (see https://www.mediawiki.org/wiki/API:Block)
+
+         Raises:
+             ValueError: username and userid params are both not set or set
+             APIError (mwclient.errors.APIError): API errors (see codes available
+                       on https://www.mediawiki.org/wiki/API:Block#Possible_errors)
+        """
+        return self._block_unblock_user(False, username, userid, reason, tags)
+
+    def _block_unblock_user(
+        self, block=True, username=None, userid=None,
+        reason=None, tags=None
+    ):
+        """
+        Blocks or unblocks a user. You should not use this protected method: prefers
+        the use of block_user or unblock_user. See docs of theses public methods.
+        """
+        if (not username and not userid) or (username and userid):
+            raise ValueError('username OR userid are required')
+        kwargs = {
+            'token': self.get_token('csrf'),
+        }
+        if username:
+            kwargs['user'] = username
+        if userid:
+            kwargs['userid'] = userid
+        if reason:
+            kwargs['reason'] = reason
+        if tags:
+            kwargs['tags'] = '|'.join(tags)
+        action = 'block' if block else 'unblock'
+        resp = self.post(action, **kwargs)
+        if resp:
+            if action == 'block' and action in resp:
+                return resp[action]
+            elif action == 'unblock' and 'id' in resp:
+                return resp
+        raise errors.APIError('unkown', 'Can not %s user' % action, kwargs=kwargs)
+
+    def get_user_groups(self, username=None, userid=None):
+        """
+         Retrieves groups the user belongs to
+
+             >>> try:
+             ...     site.get_user_groups('SomeUser')
+             ... except mwclient.errors.APIError:
+             ...     print('Can not retrieves user\'s groups: %s' % e)
+
+         Args:
+             username (str): User name concerned, OR
+             userid (int): User ID concerned
+
+         Returns:
+             List of the current groups names the user belongs to.
+
+         Raises:
+             ValueError: username and userid params are both not set or set
+             UserNotFound (mwclient.errors.UserNotFound): wanted user doest not exists
+             APIError (mwclient.errors.APIError): other API errors
+        """
+        if (not username and not userid) or (username and userid):
+            raise ValueError('username OR userid are required')
+
+        kwargs = {
+            'list': 'users',
+            'usprop': 'groups',
+        }
+        if username:
+            kwargs['ususers'] = username
+        elif userid:
+            kwargs['ususerids'] = userid
+
+        resp = self.get('query', **kwargs)
+
+        if 'missing' in resp['query']['users'][0]:
+            raise errors.UserNotFound(code='missing', info=None, kwargs=kwargs)
+        return resp['query']['users'][0].get('groups', [])
```

Comment:
```suggestion
        user = self.get_user(username, userid, 'groups')

        return user.get('groups', [])
```

to reduce the amount of code duplication.